client/src/components/Link/Link.tsx

```diff
@@ -28,29 +28,27 @@ const handleClick = (
   }
 };
 
-interface ButtonProps {
+interface LinkProps {
   className?: string;
   accessibleLabel?: string;
   href?: string;
   target?: string;
   preventDefault?: boolean;
   onClick?(e: React.MouseEvent): void;
-  dialogTrigger?: boolean;
   navigate?(url: string): Promise<void>;
 }
 
 /**
  * A reusable button. Uses a <a> tag underneath.
  */
-const Button: React.FunctionComponent<ButtonProps> = ({
+const Link: React.FunctionComponent<LinkProps> = ({
   className = '',
   children,
   accessibleLabel,
-  href = '#',
+  href = '/#',
```

Comment:
```suggestion
  href = '#',
```

I understand your reasoning here but we should keep this as the current default, this is because the click handler above will check for this specific value when blocking the click.

This is mostly used in the page explorer menu, our goal is to make the clicks work to 'expand' the content but also allow "ctrl + click" to open the suitable link a new window.

Sometimes there is no suitable link to ctrl+click on so we use the hash to represent this.